File: 0001_0100/0018/0018.c
#include <stdio.h>
#define N 428

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, i;
	int s[N] = {0};

	fscanf(f, "%d", &n);
	s[0] = 1;
	for (int j = 2; j <= n; ++j)
	{
		for (i = 0; i < N; ++i)
			s[i] *= j;
		for (i = 0; i < N; ++i)
			if (s[i] > 999999)
				s[i + 1] += s[i] / 1000000, s[i] %= 1000000;
	}
	for (i = N - 1; s[i] == 0; --i);
	fprintf(q, "%d", s[i]);
	while (--i >= 0)
		fprintf(q, "%06d", s[i]);
	return 0;
}

File: 0001_0100/0018/0018.go
package main

import (
	"fmt"
	"math/big"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int64
	fmt.Fscan(f, &n)
	k := big.NewInt(1)
	for ; n > 1; n-- {
		k.Mul(k, big.NewInt(n))
	}
	fmt.Fprint(q, k)
}
